src/game/shared/Multiplayer/multiplayer_animstate.cpp

```diff
@@ -1287,6 +1287,11 @@ void CMultiPlayerAnimState::UpdateGestureLayer( CStudioHdr *pStudioHdr, GestureS
 	// Get the current cycle.
 	float flCycle = pGesture->m_pAnimLayer->m_flCycle;
 	flCycle += pPlayer->GetSequenceCycleRate( pStudioHdr, pGesture->m_pAnimLayer->m_nSequence ) * gpGlobals->frametime * GetGesturePlaybackRate() * pGesture->m_pAnimLayer->m_flPlaybackRate;
+	if ( flCycle < 0.0f )
```

Comment:
If blanket-applying the negative cycle handling may cause issues, this can be further guarded by also checking that playbackRate specifically is negative before resetting cycle to 1.0.
```suggestion
	float flCycle = pGesture->m_pAnimLayer->m_flCycle;
	float flPlaybackRate = GetGesturePlaybackRate();
	flCycle += pPlayer->GetSequenceCycleRate( pStudioHdr, pGesture->m_pAnimLayer->m_nSequence ) * gpGlobals->frametime * flPlaybackRate * pGesture->m_pAnimLayer->m_flPlaybackRate;
	if ( flCycle < 0.0f && flPlaybackRate < 0.0f )
```